### PR TITLE
Burgers, Kuramoto Sivashinsky and Korteweg De Vries PDEs

### PR DESCRIPTION
**Pull request by yb6599**
I added implementations of Burgers 1D, Kuramoto Sivashinsky, and Korteweg De Vries PDEs. I can run experiments on Burgers and KS, although the coefficients are innacurate. Korteweg De Vries is stuck on gen_pde_data for a long time, so it does not work yet.

### CODE REVIEW

File: src/gen_experiments/__init__.py [hunk 1]
```diff
@@ -228,11 +235,11 @@ def _convert_abs_rel_noise(grid_vals: list, grid_params: list, recent_results: d
     "pde": ParamDetails(
         ND({
             "featcls": "pde",
-            "library_functions": [identity],
-            "function_names": [identity],
+            "library_functions": [lambda x: x, lambda x: x * x],
+            "function_names": [lambda x: x, lambda x: x + x],
```
Comment (Jacob-Stevens-Haas) [inline src/gen_experiments/__init__.py:239]: `mitosis` does not allow lambda functions in `Parameter` objects.  That was why we switched to `identity`.  Running experiments in this branch hits this error.

File: src/gen_experiments/pdes.py [hunk 2]
```diff
@@ -19,6 +19,32 @@ def diffuse1D(t, u, dx, nx):
...
+
+def ks(t, u ,dx, nx):
+    u = np.reshape(u, nx)
+    u[0] = 0
+    u[-1] = 0
+    ux = SpectralDerivative(d=1, axis=0)._differentiate(u, dx)
+    uxx = SpectralDerivative(d=2, axis=0)._differentiate(u, dx)
+    uxxxx = SpectralDerivative(d=4, axis=0)._differentiate(u, dx)
+    return np.reshape(-uxx-uxxxx-u*ux, nx)
+    # return np.reshape(-uxx-uxxxx-0.5*ux*ux)
```
Comment (Jacob-Stevens-Haas) [inline src/gen_experiments/pdes.py:38]: Remove comment.

**Review by Jacob-Stevens-Haas — CHANGES_REQUESTED** [review]
Thanks for meeting Zach about this!  Try to provide a command line example of running these experiments, please.  Trying `diffuse1d` breaks for the reason mentioned below.

**Comment by Jacob-Stevens-Haas** [comment]
Also see if KDV can work if you change the integration tolerances in `gen_pde_data`

**Comment by yb6599** [comment]
Hey Jake, the experiments for Burgers and Kuramoto Sivashinsky Equations should work now.
For Burgers, try `python -m gen_experiments burgers1D --seed 19 --param diff_params=test_axis --param feat_params=pde2 --param opt_params=test`
For KS, try `python -m gen_experiments ks --seed 19 --param diff_params=test_axis --param feat_params=pde2 --param opt_params=test`

**Pull request merged**